### PR TITLE
refactor(restaurant-models): change the type of Restaurant models fie…

### PR DESCRIPTION
**Pull request by SodokinMarius**
…lds into FloatField

### CODE REVIEW

**Comment by koladev32** [comment]
- Le nom du PR n'a pas été modifié

**Comment by koladev32** [comment]
Et ce refactoring concerne uniquement les restaurants models non? Pourquoi ajouter les tests déjà?

**Comment by koladev32** [comment]
Il y a 4 commits et 15 fichier changés? Comment suivre cela?

File: restaurant/views.py [hunk 1]
```diff
@@ -14,8 +14,9 @@
 from django.http import Http404
```
Comment (koladev32) [inline restaurant/views.py]: This file should be deleted

File: authentification/viewsets.py [hunk 2]
```diff
@@ -1,24 +1,65 @@
...
 
-        obj = User.objects.get(lookup_field_value)
-        self.check_object_permissions(self.request, obj)
+        user = User.objects.get(lookup_field_value)
+        self.check_object_permissions(self.request, user)
+
+        return user
+
+
+class LoginViewSet(ModelViewSet):
```
Comment (koladev32) [inline authentification/viewsets.py:27]: You should precise the only allowed methods here

File: authentification/viewsets.py [hunk 3]
```diff
@@ -1,24 +1,65 @@
...
+        serializer=self.get_serializer(data=request.data)
+        
+        if serializer.is_valid() :
+            response={"message": "User connected successfully. ","data":serializer.validated_data}
+            return Response(data=response,status=status.HTTP_200_OK)
+        return  Response(data=serializer.errors,status_code=status.HTTP_400_BAD_REQUEST)
+        
+    
+   
+class SignupViewSet(ModelViewSet):
```
Comment (koladev32) [inline authentification/viewsets.py:42]: Good job

File: authentification/viewsets.py [hunk 4]
```diff
@@ -1,24 +1,65 @@
...
+    http_method_names = ['post']
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+
+        if serializer.is_valid():
+            user = serializer.save()
+            response={"message": "User created successfully. ","data":serializer.validated_data}
+            return Response(data=response,status=status.HTTP_200_OK)
+        return Response(data=serializer.errors,status=status.HTTP_400_BAD_REQUEST)
```
Comment (koladev32) [inline authentification/viewsets.py]: No need for that. serializer.is_valid(raise_exception=True) will work for that

File: authentification/viewsets.py [hunk 5]
```diff
@@ -1,24 +1,65 @@
...
+    permission_classes=[AllowAny] #<--------- Pour le moment
+    serializer_class=LoginSerializer 
+    
+    def create(self, request, *args, **kwargs):
+        serializer=self.get_serializer(data=request.data)
+        
+        if serializer.is_valid() :
+            response={"message": "User connected successfully. ","data":serializer.validated_data}
+            return Response(data=response,status=status.HTTP_200_OK)
+        return  Response(data=serializer.errors,status_code=status.HTTP_400_BAD_REQUEST)
```
Comment (koladev32) [inline authentification/viewsets.py]: You can remove this line and just call the raise_exception

File: restaurant/urls.py [hunk 6]
```diff
@@ -2,27 +2,16 @@
...
-'''router=DefaultRouter()
 
-router.register('restaurants/',views.RestaurantAPIView.as_view(),basename='restaurants')
-router.register('restaurants/<int:pk>/',views.RestaurantDetail.as_view(),basename='single-restaurant')
+routeur=SimpleRouter()
 
-urlpatterns = [
-    path('',include(router.urls)),
-]'''
+routeur.register('restaurants',RestaurantViewSet,basename='restaurants-list')
```
Comment (koladev32) [inline restaurant/urls.py]: Not a great name for the router

File: restaurant/urls.py [hunk 7]
```diff
@@ -2,27 +2,16 @@
...
-from . import views
-from rest_framework.routers import DefaultRouter 
+from .viewsets import RestaurantViewSet
+from rest_framework.routers import SimpleRouter 
 
-'''router=DefaultRouter()
 
-router.register('restaurants/',views.RestaurantAPIView.as_view(),basename='restaurants')
-router.register('restaurants/<int:pk>/',views.RestaurantDetail.as_view(),basename='single-restaurant')
+routeur=SimpleRouter()
```
Comment (koladev32) [inline restaurant/urls.py]: it will be `router` and not `routeur` as we are going in english

**Review by koladev32 — CHANGES_REQUESTED** [review]

**Review by koladev32 — CHANGES_REQUESTED** [review]
Good job in general. Correct the reviews and merge it.

**Comment by koladev32** [comment]
Merge the code

**Comment by SodokinMarius** [comment]
ok, thanks for all comments and suggestions. I've already done all.

**Pull request merged**